src/online/gameplay.go

```diff
@@ -90,30 +90,26 @@ func populateGames(multiplayerGame *structs.MultiplayerGame, games map[string]*s
 }
 
 // Analyze every user's game to generate a new multiplayer game update.
-func getNewGameUpdate(games map[string]*structs.Game) *structs.MultiplayerGameUpdate {
+func getNewGameUpdate(state string, games map[string]*structs.Game) *structs.MultiplayerGameUpdate {
+	update := structs.MultiplayerGameUpdate{
+		State:    state,
+		WinnerID: "",
+	}
```

Comment:
No need to iterate over games if the multiplayer game is still waiting
```suggestion
	}
	if state == "waiting" {
	         return &update
	}
```